lib/rules-generator.js

```diff
@@ -332,7 +380,11 @@ util.methods(Generator, {
     switch (exp.type) {
```

Comment:
That's what I had before - but some early exit's via break.  Each case here is really an independent  "function" - which I think the internal returns make clear.  I think the distance shared return is not good for readability.

Other options?